### ISSUE TITLE
Relm4 re-exports other crates by default

### ISSUE THREAD

**Issue by Ayush1325** [Relm4/Relm4#378]
Currently, relm4 seems to re-export the following creates without any feature flag:
1. [once_cell](https://relm4.org/docs/next/once_cell/index.html)
2. [async_trait](https://relm4.org/docs/next/async_trait/index.html)
3. [tokio](https://relm4.org/docs/next/tokio/index.html)

I'm not quite sure why this is being done. While the [gtk](https://relm4.org/docs/next/gtk4/index.html) crate is also re-exported, it makes sense due to the relationship between relm4 and gtk.

However, the other 3 seem to be internal details, and it is probably possible to get rid of at least `once_cell` and `async_trait` once those features stabilize upstream. So I think they should not be re-exported without any good reason.

PS: I did check making them internal `pub(crate)` and it doesn't cause any problems at least in building

**Comment by AaronErhardt** [comment]
async_trait is needed to make `#[relm4::component(async)]` possible where relm4-macros uses the re-export to implement the async trait (see async-component example).
Tokio is used in quite a few return types in Relm4 so it makes sense to re-export it as well. Once_cell might be just for convenience. The question is whether those re-exports should be removed even if they are not needed. They can at least reduce the amount of dependencies in Cargo.toml and are automatically shipped in a version that's compatible with Relm4. And once some of those traits become part of std, we can remove them in the next major semver release.

**Comment by Ayush1325** [comment]
If the macro needs it, then `async_trait` should be there. I just think it might be good to add a comment about it.

As for the other 2, I'm just going to state my personal opinion. In the case of `once_cell` I would simply use the `std` stuff if I didn't mind using nightly (which I don't for the most part). Even if I did want to use `once_cell` crate, I would still prefer to specify it in `Cargo.toml` so that I can lock the particular version I want. I would prefer not to have to worry about `once_cell` stuff breaking just because I upgraded my `relm4` package.

The above problem is much more evident in case of something like `tokio` because it has a lot of features as well, which I will probably need to specify in the `Cargo.toml` file in any production app.

Basically, I personally prefer specifying any direct dependency I'm using in my project `Cargo.toml`, even if it is a bit more verbose.

**Comment by AaronErhardt** [comment]
I understand what you mean. It basically boils down to convenience vs. explicitness. Some are happy to have less unnecessary imports of traits that are compiled anyway, can't be imported. Some prefer to specify everything manually. I think this is mostly about your personal preference in the end.

Probably, dropping the once_cell re-export is good because it has no connection to Relm4 directly. Tokio however is an important part of the runtime, so currently I'm favour of keeping its re-export.

**Comment by Ayush1325** [comment]
Sure, I just wanted to start a discussion about them since it is based so much on personal opinion.
